lib/mail/fields/content_type_field.rb

```diff
@@ -144,10 +144,9 @@ def sanatize( val )
 
       # TODO: check if there are cases where whitespace is not a separator
       val = val.
+        gsub(/\s[\w\-]+:\s.*\\r\\n$/, ''). # cases with botched fields like Content-Type: text/html Content-Transfer-Encoding: 8bit\r\n
```

Comment:
> All I am saying is that throwing an Exception is not an answer to my quest. If an Exception is raised, the caller simply has no option but to fail the processing completely. The caller cannot fixup-and-proceed-from-where-it-failed, because there is no way of knowing where in the email body processing has failed and in what way. So I cannot fix this problem by acrobatics in my code or by fixing up the email body.

I'm sympathetic to this and I've encountered the same pattern. I'd often prefer to have an unparsable field marked with an error rather than raise an exception. That defers the issue from when the email is parsed to when the Mail user attempts to access the field, having expected it to be parsed.

Curious, if you are free to share: Are you implementing a mailing list or remailer? Or a mail filter/firewall?